src/charm.py

```diff
@@ -649,6 +676,70 @@ def _peers(self) -> Relation:
         """
         return self.model.get_relation(PEER)
 
+    def push_tls_files_to_workload(self, container: Container = None) -> None:
+        """Uploads TLS files to the workload container."""
+        if container is None:
+            container = self.unit.get_container("postgresql")
+
+        key, ca, cert = self.tls.get_tls_files()
+        if key is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_KEY_FILE}",
+                key,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+        if ca is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_CA_FILE}",
+                ca,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+        if cert is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_CERT_FILE}",
+                cert,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+
+        self.update_config()
+
+    def _restart(self, _) -> None:
+        """Restart PostgreSQL."""
+        try:
+            self._patroni.restart_postgresql()
+        except RetryError as e:
+            logger.error("failed to restart PostgreSQL")
+            self.unit.status = BlockedStatus(f"failed to restart PostgreSQL with error {e}")
+
+    def update_config(self) -> None:
+        """Updates Patroni config file based on the existence of the TLS files."""
+        enable_tls = False
+        key, ca, cert = self.tls.get_tls_files()
+        if None not in [key, ca, cert]:
+            enable_tls = True
+
+        # Update and reload configuration based on TLS files availability.
+        self._patroni.render_patroni_yml_file(enable_tls=enable_tls)
+        if not self._patroni.member_started:
+            return
+
+        restart_postgresql = enable_tls != self.postgresql.is_tls_enabled()
```

Comment:
nit, I think it would be more obviously parsable to have boring `if`'s here.

```python
if self.postgresql.is_tls_enabled() != enable_tls:
	self.on[self.restart_manager.name].acquire_lock.emit()
```

Reply:
In this specific part of the code, the check must be made before `self._patroni.reload_patroni_configuration()` is called (because it enables TLS, which is a step needed before restarting PostgreSQL).

`self._patroni.reload_patroni_configuration()` is called before the if because it is also needed when other configuration change.

Is it ok to keep the check this way due to that needs?